bcda/auth/backend.go

```diff
@@ -261,22 +226,48 @@ func getPublicKey() *rsa.PublicKey {
 	return rsaPub
 }
 
-func getJWTClaims(backend *JWTAuthenticationBackend, tokenString string) jwt.MapClaims {
+func (backend *JWTAuthenticationBackend) GetJWTClaims(tokenString string) jwt.MapClaims {
+	token, err := backend.GetJWTToken(tokenString)
+
+	if err != nil {
+		panic(err)
+	}
+
+	if !token.Valid {
+		return nil
+	}
+
+	return token.Claims.(jwt.MapClaims)
+}
+
+func (backend *JWTAuthenticationBackend) GetJWTToken(tokenString string) (*jwt.Token, error) {
+
 	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
 		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
 			return nil, fmt.Errorf("Unexpected signing method: %v", token.Header["alg"])
 		}
 
 		return backend.PublicKey, nil
 	})
+	return token, err
+}
 
+// Save a token to the DB for a user
+func (backend *JWTAuthenticationBackend) CreateToken(user bcdagorm.User) (bcdagorm.Token, error) {
```

Comment:
What's the plan for this function? Should this be called when running the `bcda create-token` command? I only see it invoked in test cases. I'm probably just missing something.

Reply:
I don't know exactly when this will be run.  I feel like we are still waiting for approval on our auth-X schemes.

Reply:
I didn't want to change the function of this command.  I am happy to change it, but I wasn't sure if it was in scope or not.